### PR TITLE
ci: Add hedera-services -> hedera-protobufs auto-update functionality

### LINKED ISSUE #15453: Update workflows to publish and tag protobufs
We have the `hedera-protobufs-automation` service account in place in `hedera-protobufs` repo. 

Need to update the `node-flow-deploy-release-artifact.yaml` to checkout the main branch of github.com/hashgraph/protobufs in the following subdirectories
- platform
- services
- streams
- block

For each tag vA.B.C in hedera-services deployed to previewnet, this subset of protobufs will be automatically copied to the hedera-protobufs repo and the same tag vA.B.C created there on the main branch

Update the node-flow-deploy-release-artifact.yaml pipeline to include a step that checks out protobufs main, updates the folders owned by Services with (e.g. `rm -rf services/*`   and `cp -R ${SERVICES_ROOT}/hapi/hedera-protobufs/services/* services/`; then creates the target Services tag in the protobufs repo.

### PR DESCRIPTION
**Pull request by mishomihov00**
**Description**:

Updates the workflow in a way that when a new tag is pushed in the hedera-services repo, it will checkout the hedera-protobufs repo, update the files there, push directly to main in the protobufs repo and create the same tag as the one in the hedera-services.

**Related issue(s)**:

Fixes #15453 

**Notes for reviewer**:

Test workflow run: https://github.com/hashgraph/hedera-services/actions/runs/11738548571/job/32701293490

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: .github/workflows/node-flow-deploy-release-artifact.yaml [hunk 1]
```diff
@@ -173,9 +173,67 @@ jobs:
...
-            "msg": "${{ inputs.msg }}",
-            "sha": "${{ inputs.sha }}"
-          }'
+          inputs: |
+            {
+              "ref": "${{ inputs.ref }}",
+              "author": "${{ inputs.author }}",
+              "msg": "${{ inputs.msg }}",
+              "sha": "${{ inputs.sha }}"
+            }
```
Comment (rbarkerSL) [inline .github/workflows/node-flow-deploy-release-artifact.yaml:182]: Why did this need to be updated?

Comment (mishomihov00) [reply .github/workflows/node-flow-deploy-release-artifact.yaml]: It was changed due to formatting errors. It turned out that only a single tab was needed.

File: .github/workflows/node-flow-deploy-release-artifact.yaml [hunk 2]
```diff
@@ -174,8 +174,65 @@ jobs:
...
+      - name: Import GPG key for commit signoff
+        id: gpg_import
+        uses: crazy-max/ghaction-import-gpg@cb9bde2e2525e640591a934b1fd28eef1dcaf5e5 # v6.2.0
+        with:
+          gpg_private_key: ${{ secrets.SVCS_GPG_KEY_CONTENTS }}
+          passphrase: ${{ secrets.SVCS_GPG_KEY_PASSPHRASE }}
+          git_user_signingkey: true
+          git_commit_gpgsign: true
+          git_tag_gpgsign: true
+
```
[the remark this answers is not in the document]
Comment (rbarkerSL) [reply .github/workflows/node-flow-deploy-release-artifact.yaml:228]: I think it just controls how the gpg key can be used. This seems okay to me.

[the remark this answers is not in the document]
Comment (mishomihov00) [reply .github/workflows/node-flow-deploy-release-artifact.yaml:228]: Yeah, it controls the GPG key usage. See here: https://github.com/step-security/ghaction-import-gpg?tab=readme-ov-file#inputs

File: .github/workflows/node-flow-deploy-release-artifact.yaml [hunk 3]
```diff
@@ -174,8 +174,65 @@ jobs:
...
+      - name: Update the folders owned by Services
+        working-directory: hedera-protobufs
+        run: |
+          git push --delete origin v${{ needs.prepare-tag-release.outputs.version }} || true
+          git tag --delete v${{ needs.prepare-tag-release.outputs.version }} || true
+          rsync -a --delete ../hapi/hedera-protobufs/services/ services/
+
+      - name: Import GPG key for commit signoff
+        id: gpg_import
+        uses: crazy-max/ghaction-import-gpg@cb9bde2e2525e640591a934b1fd28eef1dcaf5e5 # v6.2.0
```
[the remark this answers is not in the document]
Comment (mishomihov00) [reply .github/workflows/node-flow-deploy-release-artifact.yaml]: Done.

**Pull request merged**